### ISSUE TITLE
Bug in uglify-js leads to broken build (Invalid left-hand side expression in postfix operation)

### ISSUE THREAD

**Issue by m4r71n** [systemjs/builder#776]
I don't know exactly which code causes the error but it will be uglified to `0++`
https://github.com/mishoo/UglifyJS2/issues/1516

**Comment by arijitdasgupta** [comment]
Had the same issue.

While digging through the code found this where `systemjs-build` minifying, it can be related to the this piece

```javascript
function minify(output, fileName, mangle, uglifyOpts) {
  var uglify = require('uglify-js');
  var ast;
  try{
    ast = uglify.parse(output.source, { filename: fileName });
  } catch(e){
    throw new Error(e);
  }
  ast.figure_out_scope();
  
  ast = ast.transform(uglify.Compressor(uglifyOpts.compress));
  ast.figure_out_scope();
  if (mangle !== false)
    ast.mangle_names();

  var sourceMap;
  if (output.sourceMap) {
    if (typeof output.sourceMap === 'string')
      output.sourceMap = JSON.parse(output.sourceMap);

    var sourceMapIn = output.sourceMap;
    sourceMap = uglify.SourceMap({
      file: fileName,
      orig: sourceMapIn
    });

    if (uglifyOpts.sourceMapIncludeSources && sourceMapIn && Array.isArray(sourceMapIn.sourcesContent)) {
      sourceMapIn.sourcesContent.forEach(function(content, idx) {
        sourceMap.get().setSourceContent(sourceMapIn.sources[idx], content);
      });
    }
  }
```

**Comment by arijitdasgupta** [comment]
Related issue https://github.com/mishoo/UglifyJS2/issues/1516

**Comment by m4r71n** [comment]
Fixed with uglify-js 2.8.1
